lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -695,7 +695,10 @@ def podspec_path_from_options(options)
           podfile.defined_in_file.dirname + name
         elsif options[:autodetect]
           glob_pattern = podfile.defined_in_file.dirname + '*.podspec{,.json}'
-          Pathname.glob(glob_pattern).first
+          path = Pathname.glob(glob_pattern).first
+          raise Informative, 'Could not locate a *.podspec file in the current directory.' unless path
```

Comment:
`Could not locate a podspec in the current directory. You can specify the path via ...`